distributed/scheduler.py

```diff
@@ -4190,6 +4258,17 @@ def update_graph(
 
         dependencies = dependencies or {}
 
+        if len(tasks) > 1 or Computation._recent is None:
+            # TODO: maybe reuse old computation based on time interval?
```

Comment:
I would propose adding a ctx manager here instead of trying to be smart. something like

```python
with computation(id="my-favourite-computation"):
    while True:
        client.submit(foo)
```

Most users wouldn't need to use this since we'd assign every compute its own computation. The ctx manager would work similar to annotations, maybe the computation ID is an annotation? idk.

We can iterate on this, of course. I would like to avoid any fragile timing based heuristics if possible

Reply:
Not applying heuristics here would also allow us to not store state. I believe there wouldn't be a reason for the `_recent` if we made it explicit

Reply:
My current thinking is that we continue to use the previous computation until it finishes.  Once it finishes then we close it out and start a new one.  

So the following would probably result in a single computation

```python
x = x.persist()
y = y.persist()
z = (x + y).sum().compute()
```

Reply:
We now use the previous computation if `Scheduler.total_occupancy > epsilon`, so there is no longer a magic time interval.

I do think that a magic time interval might at some point be welcome, but I'm happy to wait to feel the need for that.